src/event_pusher/mod_event_pusher_http.erl

```diff
@@ -113,33 +114,19 @@ make_req(Acc, Dir, Domain, Sender, Receiver, Message, Opts) ->
                 sender => Sender, receiver => Receiver, direction => Dir,
                 server => Domain, pool_name => PoolName, acc => Acc},
     ?LOG_INFO(LogMeta),
-    T0 = os:timestamp(),
-    {Res, Elapsed} = case mongoose_http_client:post(HostType, PoolName, Path, Headers, Body) of
-                         {ok, _} ->
-                             {ok, timer:now_diff(os:timestamp(), T0)};
-                         {error, Reason} ->
-                             {{error, Reason}, 0}
-                     end,
-    record_result(HostType, Res, Elapsed, LogMeta),
-    ok.
-
-ensure_metrics(HostType) ->
-    mongoose_metrics:ensure_metric(HostType, ?SENT_METRIC, spiral),
-    mongoose_metrics:ensure_metric(HostType, ?FAILED_METRIC, spiral),
-    mongoose_metrics:ensure_metric(HostType, ?RESPONSE_METRIC, histogram),
-    ok.
-
-record_result(HostType, ok, Elapsed, _LogMeta) ->
-    mongoose_metrics:update(HostType, ?SENT_METRIC, 1),
-    mongoose_metrics:update(HostType, ?RESPONSE_METRIC, Elapsed),
-    ok;
-record_result(HostType, {error, Reason}, _, LogMeta) ->
-    mongoose_metrics:update(HostType, ?FAILED_METRIC, 1),
+    mongoose_instrument:span(?SENT_METRIC, #{host_type => HostType},
+                             fun mongoose_http_client:post/5, [HostType, PoolName, Path, Headers, Body],
+                             fun(Time, Result) -> handle_post_result(Time, Result, LogMeta) end).
+
+handle_post_result(Time, {ok, {CodeBin, _Body}}, #{sender := Sender}) ->
+    % Fusco returns HTTP response codes as binaries
+    Code = binary_to_integer(CodeBin),
```

Comment:
The only problem I see is that fusco seems to just return the three characters, while some misbehaving server might return a non-numerical status code. In this case, maybe instrumentation should't fail? Maybe we could return the binary?

Reply:
I also wasn't sure how to tackle it. I will change this to return the binary, as indeed the instrumentation shouldn't fail.